feil-behandler/src/main/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/feilbehandler/river/FeilLytter.kt

```diff
@@ -46,25 +49,61 @@ class FeilLytter(rapidsConnection: RapidsConnection, private val repository: Bak
     override fun onPacket(packet: JsonMessage, context: MessageContext) {
         sikkerLogger.info("Mottok feil: ${packet.toPretty()}")
         val fail = toFailOrNull(packet.toJson().parseJson().toMap())
+        if (fail == null) {
+            sikkerLogger.warn("Kunne ikke parse feil-objekt, ignorerer...")
+            return
+        }
         if (skalHaandteres(fail)) {
-            sikkerLogger.info("Lagrer mottatt pakke!")
-            val jobb = Bakgrunnsjobb(
-                type = jobbType,
-                data = fail?.utloesendeMelding.toString()
-            )
-            lagreJobb(jobb)
+            // slå opp transaksjonID. Hvis den finnes, kan det være en annen feilende melding i samme transaksjon (forskjelig behov): Lagre i så fall
+            // med egen id. Denne id vil så sendes med som ny transaksjonID ved rekjøring..
+            val jobbId = fail.transaksjonId
+            val eksisterendeJobb = repository.getById(jobbId)
+            if (eksisterendeJobb != null) {
+                if (fail.utloesendeMelding.toMap()[Key.BEHOV] != eksisterendeJobb.data.parseJson().toMap()[Key.BEHOV]) {
+                    sikkerLogger.info("Id $jobbId finnes fra før med annet behov. Lagrer en ny jobb.")
+                    lagre(Bakgrunnsjobb(type = jobbType, data = fail.utloesendeMelding.toString()))
+                } else {
+                    oppdater(eksisterendeJobb)
+                }
+            } else {
+                sikkerLogger.info("Lagrer mottatt pakke!")
+                val jobb = Bakgrunnsjobb(
+                    uuid = fail.transaksjonId,
+                    type = jobbType,
+                    data = fail.utloesendeMelding.toString()
+                )
+                lagre(jobb)
+            }
         }
     }
 
-    private fun lagreJobb(jobb: Bakgrunnsjobb) {
-        repository.save(jobb)
+    private fun oppdater(jobb: Bakgrunnsjobb) {
+        // Dette må gjøres her fordi jobbene er asynkrone og bakgrunnsjobbService ikke får vite at jobben feiler i disse tilfellene
+        // BakgrunnsjobbService finnVentende() tar heller ikke hensyn til forsøk, kun status på jobben!
+        if (jobb.forsoek > jobb.maksAntallForsoek) {
+            jobb.status = BakgrunnsjobbStatus.STOPPET
+            sikkerLogger.warn("Maks forsøk nådd, stopper jobb med id ${jobb.uuid} permanent!")
+        } else {
+            jobb.status = BakgrunnsjobbStatus.FEILET
+        }
```

Comment:
Har bakgrunnsjobbene noe konsept om vellykket rekjøring?

Reply:
Nja, bakgrunnsjobbService plukker jobber, øker forsøk+1 og setter status til OK før den rekjører. 
I bgjobb-biblioteket finnes en catch på prosesser(jobb) og der settes jobb til FEILET , men iom at i vår verden gjøres "fire-and-forget", så vil ikke det slå til, med mindre vi ikke klarer å sende ut på rapid. 
Så i vår asynkrone verden er rekjøring nesten alltid vellykket, inntil vi plutselig ser en "ny" feil som matcher vår orginale feil - dette er tegnet på at jobben igjen har status FEILET :)